### DISCUSSION TITLE
allure-steps are not generating

### DISCUSSION THREAD

**Discussion by pattaswamyr** [allure-framework/allure2#2048] [Questions & Support]
Hi Team,

We are using allure-reports in out C#-Nunit framework, but problem is allure-steps are not generating in properly.
When we run the same framework on multiple machines, in one machine it's generating allure-steps in another machine it's not.

Even same machine where, it generate the steps also have intermittent issue, if remove allure libraries and it again it will starts displaying steps.
Here is the library versions we are using :
C# .Net  :  4.7.2 
NUnit: 3.13.3
NUnit.ConsoleRunner  : 3.13
NUnit.Allure: 1.2.1.1
Allure.Steps: 1.0.9
Even we tried with new allure-reports  (i.e. Allure.Nunit 2.9.5.-preview.1 ).

Can you please help how to fix this issue. to generate the reports along with steps. 

Thanks
Raghu

**Comment by neparij** [comment]
Hello @pattaswamyr! We need a little bit more info to find a way to resolve your issue.
Could you provide:
- The command you launch the tests
- Which way do you create your steps: is it [AllureStep] attribute or the WrapInStep method?

If you can provide some code samples (without confidential data) it may help us.

**Reply by neparij** [reply]
PS
If your tests depend on AllureLifecycle class: Allure.NUnit package is compatible with Allure.Net.Commons (not the Allure.Commons one as it was before)

**Reply by pattaswamyr** [reply]
Thank you Nik,

a. We used to execute the test cases from Visual-studio directly (by right click then select RUN)
b. We are using [AllureStep("Here is steps")]  to create the steps.

**Comment by delatrie** [comment]
Hi, @pattaswamyr !
I've been trying to reproduce the effect, but no luck so far.

Since you're using the `AllureStep` attribute there might be a chance aspect injector doesn't generate the step injections properly in some circumstances.

Below are some questions and suggestions that may help us to figure out what's going on.

1. Do you have any successful test run (i.e., with steps) between the latest build and a failed test run (i.e., without the steps)?
1. What's the OS and the processor architecture of the machine you run your tests on?
1. Does the issue occur if you use `WrapInStep` instead of `[AllureStep]`? To check it, use this:

    ```csharp
    [Test]
    public void MyTest()
    {
        AllureLyfecycle.Instance.WrapInStep(() => {}, "MyStep");
    }
    ```

    instead of this:

    ```csharp
    [Test]
    public void MyTest()
    {
        this.MyStep();
    }

    [AllureStep]
    void MyStep(){}
    ```
1. What does aspect injector tells you during the latest build that precedes a test run with missing steps? There should be a line that looks like this:
    ```
    AspectInjector|2.8.1: Found <x> aspects, <y> injections
    ```
    Do you have a line like this? What values of `<x>` and `<y>` do you get? Does the number aspect injector reports to you matches the actual number of injections in your test project (i.e., the total number of `AllureBefore`, `AllureAfter` and `AllureStep` attributes used? Please, provide me also with any additional info reported by aspect injector: warnings, errors, etc, if there is some.
